ex_app/lib/main.py

```diff
@@ -148,6 +154,285 @@ async def proxy_frontend_requests(request: Request, path: str):
         return response_to_nc
 
 
+async def sync_timer():
+    while True:
+        if app_enabled:
+            print("Running workflow sync")
+            workspace = 'YOUR_WORKSPACE'
+            token = 'YOUR_TOKEN_HERE'
+            flow_paths = await get_flow_paths(workspace, token)
+            expected_listeners = await get_expected_listeners(workspace, token, flow_paths)
+            print(json.dumps(expected_listeners, indent=4), flush=True)
+            for expected_listener in expected_listeners:
+                registered_listeners = await get_registered_listeners_for_uri(expected_listener['webhook'])
+                for event in expected_listener['events']:
+                    listener = next(filter(lambda listener: listener['event'] == event, registered_listeners), None)
+                    if listener is not None:
+                        json.dumps(listener)
+                        if json.dumps(listener['eventFilter']) != json.dumps(expected_listener['filters']):
+                            await update_listener(listener, expected_listener['filters'], token)
+                    else:
+                        await register_listener(event, expected_listener['filters'], expected_listener['webhook'], token)
+            registered_listeners = await get_registered_listeners()
```

Comment:
Since you get the list of all listeners anyway, does it make sense to fetch them by uri above?
You could fetch the list first and search in there.